internal/application/command.go

```diff
@@ -195,6 +207,35 @@ func readDeviceCommand(device models.Device, commandName string, attributes stri
 	return res, nil
 }
 
+// readRegexDeviceResources skips device profile check and leverage ProtocolDriver implementation directly
+func readRegexDeviceResources(device models.Device, resourceName string, attributes string, dic *di.Container) (res *dtos.Event, edgexErr errors.EdgeX) {
```

Comment:
This should not depend on the Protocol Driver implementation. It should apply the RegEx to the list of Resource/Command names and return the list of match items. 

Question is does this apply to device commands or just device resources.

Reply:
Applying to just device resources makes more sense to me since device command is capable of aggregating device resources.
Also applying on both will:
1. need to check both device resource and device command against RegEx in device profile
2. possibly returning duplicate device resource (or check ResourceOperations between matched DeviceCommand and matched DeviceResource to avoid this)

@lenny-intel @iain-anderson @cloudxxx8 thought ?

Reply:
Would suggest matching on DeviceResources only

Reply:
In the Mar-22-2023 Core WG meeting, we agree with the design suggested by @iain-anderson